app/main.py

```diff
@@ -1,2 +1,67 @@
+from typing import Any, Hashable, List, Optional
+from fractions import Fraction
+
+
+class Node:
```

Comment:
Move this class inside of Dictionary - since it is part of it internal implementation